packages/block-editor/src/components/block-binding-support/index.js

```diff
@@ -0,0 +1,49 @@
+/**
+ * WordPress dependencies
+ */
+import { addFilter } from '@wordpress/hooks';
+
+/**
+ * Internal dependencies
+ */
+import withBlockBindingSupport from './with-block-binding-support';
+
+export const BLOCK_BINDINGS_ALLOWED_BLOCKS = {
+	'core/paragraph': [ 'content' ],
+	'core/heading': [ 'content' ],
+	'core/image': [ 'url', 'title', 'alt' ],
+	'core/button': [ 'url', 'text', 'linkTarget' ],
+};
+
+export function isItPossibleToBindBlock( blockName ) {
+	return blockName in BLOCK_BINDINGS_ALLOWED_BLOCKS;
+}
+
+export default function extendBlockWithBoundAttributes( settings, name ) {
+	if ( ! isItPossibleToBindBlock( name ) ) {
+		return settings;
+	}
+
+	return {
+		...settings,
+		/*
+		 * Expose relevant data through
+		 * the block context.
+		 */
+		usesContext: [
+			...new Set( [
+				...( settings.usesContext || [] ),
+				'postId',
+				'postType',
+				'queryId',
+			] ),
+		],
+		edit: withBlockBindingSupport( settings.edit ),
+	};
+}
+
+addFilter(
```

Comment:
I believe the whole logic should probably be inside the `block-editor/src/hooks` folder as we are doing right now with the [use-bindings-attributes](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-editor/src/hooks/use-bindings-attributes.js). It is not a component we are exposing to use if I am not mistaken.